test/gurka/test_filter.cc

```diff
@@ -0,0 +1,108 @@
+#include "baldr/graphreader.h"
+#include "mjolnir/util.h"
+
+#include "gurka.h"
+#include "test/test.h"
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::gurka;
+
+valhalla::gurka::map the_map = {};
+
+class FilterData : public ::testing::Test {
+protected:
+  valhalla::gurka::map BuildPBF(const std::string& workdir) {
+    constexpr double gridsize_metres = 100;
+
+    const std::string ascii_map = R"(
+		A--------B--------C
+                 |
+                 |
+                 |
+                 D
+    )";
+
+    const gurka::ways ways = {
+        {"ABC",
+         {{"highway", "residential"},
+          {"name:forward", "name in forward dir"},
+          {"name:backward", "name in backward dir"},
+          {"osm_id", "100"}}},
+        {"BD", {{"highway", "footway"}}},
+    };
+
+    if (!filesystem::exists(workdir)) {
+      bool created = filesystem::create_directories(workdir);
+      EXPECT_TRUE(created);
+    }
+
+    constexpr double gridsize = 100;
+
+    const auto layout =
+        gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {-82.68811, 40.22535});
+
+    auto pbf_filename = workdir + "/map.pbf";
+    detail::build_pbf(layout, ways, {}, {}, pbf_filename);
+
+    valhalla::gurka::map result;
+    result.nodes = layout;
+    return result;
+  }
+};
+
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(FilterData, CheckStreetNamesAfterFilter) {
+
+  const std::string workdir = "test/data/gurka_filter";
+
+  if (!filesystem::exists(workdir)) {
+    bool created = filesystem::create_directories(workdir);
+    EXPECT_TRUE(created);
+  }
+
+  the_map = BuildPBF(workdir);
+
+  const std::string sqlite = {VALHALLA_SOURCE_DIR "test/data/language_admin.sqlite"};
+  the_map.config =
+      test::make_config(workdir,
+                        {{"mjolnir.admin", {VALHALLA_SOURCE_DIR "test/data/language_admin.sqlite"}},
+                         {"mjolnir.tile_dir", workdir + "/tiles"},
+                         {"mjolnir.include_pedestrian", "false"}});
```

Comment:
ok so the basic gist of this test is that by removing the footway edge, the filter was setting the residential edges to all have the same name when they should have had different names in the forward and reverse directions. a comment mentioning this would be good :smile: also too bad we cant add this to an existing tile building test suite, we have way too many executables for the tests

Reply:
aha, thanks, didn't see this comment! I still think the tile building could be done in the more usual/terse way, not so elaborate, no?

Reply:
Adding a comment.

Reply:
Also, I hope to add more tests for aggregating the edges when we can combine them.